app/src/main/java/de/westnordost/streetcomplete/osm/surface/SurfaceItem.kt

```diff
@@ -0,0 +1,98 @@
+package de.westnordost.streetcomplete.osm.surface
+
+import android.content.res.Resources
+import de.westnordost.streetcomplete.R
+import de.westnordost.streetcomplete.osm.surface.Surface.*
+import de.westnordost.streetcomplete.view.DrawableImage
+import de.westnordost.streetcomplete.view.ResImage
+import de.westnordost.streetcomplete.view.ResText
+import de.westnordost.streetcomplete.view.RotatedCircleDrawable
+import de.westnordost.streetcomplete.view.controller.StreetSideDisplayItem
+import de.westnordost.streetcomplete.view.controller.StreetSideItem2
+import de.westnordost.streetcomplete.view.image_select.DisplayItem
+import de.westnordost.streetcomplete.view.image_select.Item
+
+fun List<Surface>.toItems() = this.map { it.asItem() }
+fun List<Surface>.toItemsWithFakeNullPossibility() = this.map { it.asItemWithFakeNullPossibility() }
+
+fun ParsedSurface.asItem(): DisplayItem<Surface?> = when (this) {
+    is Surface -> this.asItemWithFakeNullPossibility()
+    is UnknownSurface -> this.asItem()
+}
+
+fun UnknownSurface.asItem(): DisplayItem<Surface?> = Item(null,  R.drawable.background_fully_transparent, R.string.unknown_surface_title)
+
+fun Surface.asItem(): DisplayItem<Surface> = Item(this, iconResId, titleResId)
+
+fun Surface.asItemWithFakeNullPossibility(): DisplayItem<Surface?> = Item(this, iconResId, titleResId)
```

Comment:
no idea is it possible to get rid of this absurd construct, but compiler was demanding that passing `DisplayItem<Surface>` where `DisplayItem<Surface?>` is expected is not OK.

So I created this weirdness.

Reply:
Not sure. Maybe cast it as last resort? (Or consult with kotlin documentation first)

Reply:
Kotlin documentation has not helped, but after I joined official Kotlin Slack and asked question I got several hints that I will investigate :)

Reply:
Partially resolved: `DisplayItem<Surface>` allow setting `value` to `null` (is nullable inside)

But `StreetSideDisplayItem<Surface>` does not allow setting `value` to `null` (is NOT nullable inside, apparently due to showing latest answer for applying). I will try to rework it to start being nullable inside.